### PR TITLE
Update OP ATST to allow for value scoring

### PR DESCRIPTION
**Pull request by wslyvh**
The previous implementation required that attestation matched specific key/value pairs.

This update allows to use the attestation values if these are numeric. Otherwise still counts as 1.
This allows for more control over the ATST data provider.

### CODE REVIEW

File: group-generators/helpers/data-providers/atst/index.ts [hunk 1]
```diff
@@ -10,23 +14,40 @@ export class AttestationStationProvider extends GraphQLProvider {
...
+    const dataProfiles: FetchedData = {};
+    for await (const item of this._getAttestations(params)) {
+      const value = isNaN(Number(item.val)) ? 1 : Number(item.val);
+      dataProfiles[item.receiver] = value;
+    }
+
+    return dataProfiles;
+  }
+
+  public async GetAttestationValueParams(
```
[the remark this answers is not in the document]
Comment (wslyvh) [reply group-generators/helpers/data-providers/atst/index.ts]: Oops. My bad. That should've been the 'Count' function. Updated!

File: group-generators/helpers/data-providers/atst/types.ts [hunk 2]
```diff
@@ -11,13 +13,20 @@ export type GetAttestationDataType = {
 };
 
 export type GetAttestationParams = {
+  creator: string;
+  key?: string;
```
[the remark this answers is not in the document]
Comment (wslyvh) [reply group-generators/helpers/data-providers/atst/types.ts:17]: The 'optional' parameters allow for more flexibility and control over the queries. 

The idea here is that the most likely queries you'd perform on ATST are:
- Search by creator (e.g. find any attestations that I created)
These would typically give you a score of '1'
- Search by creator + key (e.g. find any attestations that I created for a specific key)
In this case, I can try to parse the (numeric) values of the attestation as a scoring mechanism. Defaults to '1'
- Search by creator + key + value (e.g. find any attestations that I created for a specific key/value pair)
These would typically give you a score of '1' again

I actually forgot the add the owner initially thinking the same. But anyone can create an attestation using the same key. The combination of owner + key is what makes each unique. 

I added some additional entities and examples queries to the subgraph as well if you want to play around :)
https://thegraph.com/hosted-service/subgraph/wslyvh/optimism-atst

[the remark this answers is not in the document]
Comment (MartinGbz) [reply group-generators/helpers/data-providers/atst/types.ts:17]: Okay I see, very clear explanation, thanks!

File: group-generators/helpers/data-providers/atst/interface-schema.json [hunk 3]
```diff
@@ -4,26 +4,55 @@
...
     {
-      "name": "Get attestation receivers",
+      "name": "Get receivers",
       "functionName": "getAttestations",
       "countFunctionName": "getAttestationsCount",
       "description": "Returns all the receivers of a specific attestation on AttestationStation",
       "args": [
+        {
+          "name": "account",
+          "argName": "account",
```
Comment (MartinGbz) [inline group-generators/helpers/data-providers/atst/interface-schema.json:14]: I think here is "creator" (sorry I haven't see this in the first review, my bad)

File: group-generators/helpers/data-providers/atst/interface-schema.json [hunk 4]
```diff
@@ -4,26 +4,55 @@
...
+    },
+    {
+      "name": "Get attestation values",
+      "functionName": "getAttestationValues",
+      "countFunctionName": "getAttestationValuesCount",
+      "description": "Uses the values of the attestations if these are numeric. Otherwise counts as 1.",
+      "args": [
+        {
+          "name": "account",
+          "argName": "account",
```
Comment (MartinGbz) [inline group-generators/helpers/data-providers/atst/interface-schema.json:43]: Same here: "creator"

**Review by MartinGbz — COMMENTED** [review]
a last little fix to do :)

**Comment by wslyvh** [comment]
Oops! Good catch. Updated

**Pull request merged**